src/navigation/index.js

```diff
@@ -48,8 +57,13 @@ const Navigator = ({ routingInstrumentation }) => {
       handleAppStateChange
     );
 
+    Linking.addEventListener('url', link => {
+      handleDeepLink(link.url);
+    });
+
     return () => {
       subscription.remove();
+      Linking.removeAllListeners();
```

Comment:
instead of removing all listeners, remove the one added before, using `remove` on the EventSubscription from `addEventListener `

Reply:
something like this
```
const deepLinkListener = Linking.addEventListener('url', link => {
      handleDeepLink(link.url);
    });

return () => {
      subscription.remove();
      deepLinkListener.remove();
}
```